### ISSUE TITLE
The client is using an unsupported version of the Socket.IO or Engine.IO protocols

### ISSUE THREAD

**Issue by fx02** [cs01/gdbgui#366]
**Describe the bug**
After opening the http://localhost:5000 I'm unable to load binary.

**To Reproduce**
Try to load binary

**Expected behavior**
It should be able to load binary.

**Screenshots**
![image](https://user-images.githubusercontent.com/2067180/101643589-922c0180-3a34-11eb-9cc1-46d28549a116.png)
![image](https://user-images.githubusercontent.com/2067180/101643628-9d7f2d00-3a34-11eb-99db-a4f28d96b173.png)

**Please complete the following information:**
* OS: CentOS7
* gdbgui version: 0.14.0.1
* gdb version (`gdb -v`): GNU gdb (GDB) Red Hat Enterprise Linux 7.6.1-119.el7
* browser [e.g. chrome, safari]: Google Chrome Version 87.0.4280.88 (Official Build) (64-bit)
* python packages (`pip freeze`):
backports.ssl-match-hostname==3.5.0.1
chardet==2.2.1
configobj==4.7.2
decorator==3.4.0
ethtool==0.8
iniparse==0.4
ipaddr==2.1.11
ipaddress==1.0.16
kitchen==1.1.1
lxml==3.2.1
M2Crypto==0.21.1
Magic-file-extensions==0.2
pciutils==1.7.3
perf==0.1
pycurl==7.19.0
pygobject==3.22.0
pygpgme==0.3
pyinotify==0.9.4
pyliblzma==0.5.3
pyOpenSSL==0.13.1
python-dateutil==1.5
python-dmidecode==3.10.13
python-linux-procfs==0.4.9
pyudev==0.15
pyxattr==0.5.1
rhnlib==2.5.65
schedutils==0.4
six==1.9.0
slip==0.4.0
slip.dbus==0.4.0
subscription-manager==1.24.26
syspurpose==1.24.26
urlgrabber==3.10
yum-metadata-parser==1.1.4

**Additional context**

**Comment by dwrobel** [comment]
I'm observing exactly the same error:
```
Opening gdbgui with default browser at http://127.0.0.1:5000
View gdbgui dashboard at http://127.0.0.1:5000/dashboard
exit gdbgui by pressing CTRL+C
The client is using an unsupported version of the Socket.IO or Engine.IO protocols (further occurrences of this error will be logged with level INFO)
(2020-12-10 22:14:31,636) The client is using an unsupported version of the Socket.IO or Engine.IO protocols (further occurrences of this error will be logged with level INFO)
```
```
$ gdbgui --version
0.14.0.1
$ gdb -v
GNU gdb (GDB) Fedora 9.1-6.fc32
```

**Comment by karlosp** [comment]
Same here:
OS: RHEL 7

$ gdbgui --version
0.14.0.1

GNU gdb (GDB) 8.3

multiple browsers:
firefox, chromium

python 3.8

**Comment by lslz627** [comment]
Same here:
OS: CentOS Linux release 7.7.1908 (Core)

$ gdbgui --version
0.14.0.1

GNU gdb (GDB) Red Hat Enterprise Linux 7.6.1-115.el7

multiple browsers:
firefox, chrome

Python 3.7.4

**Comment by twick00** [comment]
This is due to SocketIO [recently releasing version 3](url).
> TL;DR: due to several breaking changes, a v2 client will not be able to connect to a v3 server (and vice versa)


I'm not exceptionally good at python but when I `pip3 install -r requirements.txt` `pip freeze` gives me this:
```
bidict==0.21.2
Brotli==1.0.9
click==7.1.2
dnspython==2.0.0
eventlet==0.25.2
Flask==0.12.5
Flask-Compress==1.8.0
Flask-SocketIO==2.9.6
-e git+https://github.com/cs01/gdbgui/@531d89890c0b4bd3bbf15d266b9ec25a2c7eebaa#egg=gdbgui
gevent==1.5.0
gevent-websocket==0.10.1
greenlet==0.4.16
itsdangerous==1.1.0
Jinja2==2.11.2
MarkupSafe==1.1.1
monotonic==1.5
pygdbmi==0.10.0.0
Pygments==2.7.3
python-engineio==4.0.0
python-socketio==5.0.3
six==1.15.0
stdlib-list==0.8.0
Werkzeug==0.16.1
```
> `python-socketio==5.0.3`

This version will not work with the `socket.io` version defined in `package.json`:
`"socket.io": "2.0.3"`

@cs01 if you're already building a distributable binary why not just pin all the versions

**Comment by cs01** [comment]
New version 0.14.0.2 published to PyPI  thanks to @twick00's PR https://github.com/cs01/gdbgui/pull/368

> @cs01 if you're already building a distributable binary why not just pin all the versions?

The idea was to leave a range to allow for bugfixes, security fixes, etc., and allow the pip resolver to succeed if other packages are installed. But I would rather just switch to pinning everything to avoid issues like this. If you (or anyone) would like to submit a PR pinning all dependencies to requirements.txt and then reading from setup.py,  I would accept.
